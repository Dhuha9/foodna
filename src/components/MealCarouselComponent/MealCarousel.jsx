import React, { Component } from 'react';
import MealCard from '../MealCard/MealCard';
import Carousel from 'react-elastic-carousel';
import firebase from '../../firebase';
import { Link } from 'react-router-dom';
// import 'bootstrap/dist/css/bootstrap.min.css';
import '../meals.css';

const breakpoints = [
  { width: 1, itemsToShow: 1 },
  { width: 550, itemsToShow: 2, itemsToScroll: 2 },
  { width: 768, itemsToShow: 3 },
  { width: 1200, itemsToShow: 4 },
];

export default class Meals extends Component {
  constructor(props) {
    super(props);
    this.state = {
      mealsCarousel: [],
    };
  }

  componentDidMount() {
    const unsubscribe = firebase
      .firestore()
      .collection('meals')
      .onSnapshot((snap) => {
        const newMeals = snap.docs.map((doc) => ({
          id: doc.id,
          ...doc.data(),
        }));
        this.setState({
          mealsCarousel: newMeals,
        });
      });
    return () => unsubscribe();
  }

  render() {
    const { t } = this.props;

    return (
      <div>
        <div
          className="text-center vh-150 mb-5"
          style={{ backgroundColor: '#03B664' }}
        >
          <h1 className="pt-5 pb-3 text-white">
            {t('AvailableMealsComponent.headline')}
          </h1>
          <p className="pb-4 text-white fs-3" style={{ fontSize: '20px' }}>
            {t('AvailableMealsComponent.paragraph')}
          </p>
          {this.state.mealsCarousel.length === 0 ? (
            <p className="pb-4 text-yello fs-3" style={{ fontSize: '20px' }}>
              {t('AvailableMealsComponent.NoData')}
            </p>
          ) : (
            <Carousel breakPoints={breakpoints}>
              {this.state.mealsCarousel.map((e) => {
                if (e.available)
                  return (
                    <div>
                      <Link
                        to={`/recieve/${e.id}`}
                        style={{ textDecoration: 'none', color: '#000' }}
                      >
                        <MealCard
                          key={e.id}
                          img={e.image}
                          title={e.title}
                          organization={e.organization}
                          id={e.id}
                          bgColor={'buttonGray'}
                        />
                      </Link>
                    </div>
                  );
              })}
            </Carousel>
          )}
        </div>
      </div>
    );
  }
}
