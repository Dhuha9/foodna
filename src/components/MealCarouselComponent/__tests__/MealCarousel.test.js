import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MealCarousel from '../MealCarousel';
it('renders apply now correctly', () => {
  const tree = renderer.create(<MealCarousel />).toJSON();
  expect(tree).toMatchSnapshot();
});
