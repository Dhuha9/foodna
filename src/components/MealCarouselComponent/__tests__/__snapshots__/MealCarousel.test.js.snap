// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders apply now correctly 1`] = `
<div>
  <div
    className="available-meals"
  >
    <h1
      className="meals-title"
    >
      Available Meals Right Now
    </h1>
    <p
      className="meals-quote"
    >
      If you dont try this app, you wont become the hero you were meant to be
    </p>
    <div
      className="sc-jRQAMF gNESUr rec rec-carousel-wrapper "
      style={Object {}}
    >
      <div
        className="sc-eCImvq ixXTzo rec rec-carousel"
        size={
          Object {
            "height": 0,
          }
        }
        style={
          Object {
            "height": 0,
          }
        }
      >
        <button
          className="sc-bdvvaa hSiCVM rec rec-arrow rec rec-arrow-left"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❮
        </button>
        <div
          className="sc-dkPtyc imakzv rec rec-slider-container"
        >
          <div
            className="sc-hKwCoD eNbzjJ rec rec-slider"
            style={
              Object {
                "left": "0px",
                "right": "auto",
                "top": "auto",
                "transition": "all 500ms ease",
              }
            }
          >
            <div
              className="rec rec-swipable"
              onMouseDown={[Function]}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            />
          </div>
        </div>
        <button
          className="sc-bdvvaa hSiCVM rec rec-arrow rec rec-arrow-right"
          disabled={true}
          onClick={[Function]}
          style={Object {}}
          tabIndex={0}
          type="button"
        >
          ❯
        </button>
      </div>
      <div
        className="sc-iCfLBT kGrYtS rec rec-pagination"
      >
        <button
          className="sc-gKckTs cetQhm rec rec-dot rec rec-dot_active"
          onClick={[Function]}
          tabIndex={-1}
          type="button"
        />
      </div>
    </div>
  </div>
</div>
`;
